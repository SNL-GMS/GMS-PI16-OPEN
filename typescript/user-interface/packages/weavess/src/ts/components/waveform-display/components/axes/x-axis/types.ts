export interface XAxisProps {
  /** Epoch seconds end */
  endTimeSecs: number;

  /** Epoch seconds start */
  startTimeSecs: number;

  /** Add border to top */
  borderTop: boolean;

  /** Label width in px */
  labelWidthPx: number;

  /** Scrollbar width in px */
  scrollbarWidthPx: number;

  /** (optional) x-axis label  */
  label?: string;

  /** Call back to get the current view range */
  getViewRange(): [number, number];
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface XAxisState {}
