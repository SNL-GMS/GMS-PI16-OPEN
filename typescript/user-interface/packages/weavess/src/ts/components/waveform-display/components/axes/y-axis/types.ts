export interface YAxisProps {
  /** The height in percentage */
  heightInPercentage: number;

  /** Max amplitude as a number */
  maxAmplitude: number;

  /** Min amplitude as a number */
  minAmplitude: number;

  /** the y axis ticks */
  yAxisTicks?: number[];
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface YAxisState {}
