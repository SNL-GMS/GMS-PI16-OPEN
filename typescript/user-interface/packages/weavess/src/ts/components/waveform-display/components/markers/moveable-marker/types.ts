import { WeavessTypes } from '@gms/weavess-core';

export interface MoveableMarkerProps {
  /** The marker config */
  marker: WeavessTypes.Marker;

  /** The associated start marker (usually only for a selection window) */
  associatedStartMarker?: WeavessTypes.Marker;

  /** The associated end marker (usually only for a selection window) */
  associatedEndMarker?: WeavessTypes.Marker;

  /** Percentage Location 0-100 as number */
  percentageLocation: number;

  /** Label Width in px */
  labelWidthPx: number;

  /** react key but called name since key was causing fortify issues */
  name?: string;

  /** Start and end of the entire time range */
  timeRange(): WeavessTypes.TimeRange;

  /** current view port of the time range */
  viewTimeRange(): WeavessTypes.TimeRange;

  /** Returns container client width */
  containerClientWidth(): number;

  /** Returns the viewPort client width */
  viewportClientWidth(): number;

  /**
   * (optional) updates the location of the marker
   *
   * @param marker the marker
   */
  onUpdateMarker?(marker: WeavessTypes.Marker): void;

  /**  */
  updateTimeWindowSelection?(): void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface MoveableMarkerState {}
