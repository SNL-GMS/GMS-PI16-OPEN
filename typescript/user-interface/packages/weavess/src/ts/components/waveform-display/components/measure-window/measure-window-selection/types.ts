import { IconName, Intent } from '@blueprintjs/core';
import { WeavessTypes } from '@gms/weavess-core';
import * as React from 'react';

export interface MeasureWindowSelectionListenerProps {
  displayTimeRange: WeavessTypes.TimeRange;
  hotKeys: WeavessTypes.HotKeysConfiguration;
  isMeasureWindowEnabled(): boolean;
  children(measureWindowParams: {
    contentRenderer: JSX.Element;
    onMouseDown: (e: React.MouseEvent<HTMLDivElement>) => void;
  }): JSX.Element | null;
  toast(
    message: string,
    intent?: Intent | undefined,
    icon?: IconName | undefined,
    timeout?: number | undefined
  ): void;
  updateMeasureWindowPanel(
    timeRange: WeavessTypes.TimeRange,
    removeMeasureWindowSelection: () => void
  ): void;
  computeTimeSecsFromMouseXPixels(mouseXPx: number): number;
}
