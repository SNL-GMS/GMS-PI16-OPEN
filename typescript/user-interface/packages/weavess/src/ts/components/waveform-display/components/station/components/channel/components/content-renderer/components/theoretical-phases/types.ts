import { IconName, Intent } from '@blueprintjs/core';
import { WeavessTypes } from '@gms/weavess-core';

export interface TheoreticalPhasesProps {
  /** Station Id as string */
  stationId: string;

  /** The theoretical phase windows */
  theoreticalPhaseWindows: WeavessTypes.TheoreticalPhaseWindow[] | undefined;

  /** Boolean is default channel */
  isDefaultChannel: boolean;

  /** Epoch seconds start time */
  displayStartTimeSecs: number;

  /** Epoch seconds end time */
  displayEndTimeSecs: number;

  /** (optional) callback events Ex on waveform click */
  events?: WeavessTypes.ChannelContentEvents;

  /**
   * Call back for toast, which is the notification pop up
   *
   * @param message message to be displayed as a string
   * @param intent (optional) NONE, PRIMARY, WARNING, DANGER ex NONE = "none"
   * @param icon (optional) the way it looks
   * @param timeout (optional) time before message disappears
   */
  toast(message: string, intent?: Intent, icon?: IconName, timeout?: number): void;

  /**
   * Returns the time in seconds for the given clientX.
   *
   * @param clientX The clientX
   *
   * @returns The time in seconds; undefined if clientX is
   * out of the channel's bounds on screen.
   */
  getTimeSecsForClientX(clientX: number): number | undefined;

  /**
   * Toggle display of the drag indicator for this channel
   *
   * @param show True to show drag indicator
   * @param color The color of the drag indicator
   */
  toggleDragIndicator(show: boolean, color: string): void;

  /**
   * Set the position for the drag indicator
   *
   * @param clientX The clientX
   */
  positionDragIndicator(clientX: number): void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface TheoreticalPhasesState {}
