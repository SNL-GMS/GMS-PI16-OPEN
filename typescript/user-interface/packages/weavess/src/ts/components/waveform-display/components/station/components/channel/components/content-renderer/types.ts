import { IconName, Intent } from '@blueprintjs/core';
import { WeavessTypes } from '@gms/weavess-core';
import React from 'react';

import { PositionConverters } from '../../../../../../../../util/types';

export interface ContentRendererProps {
  /** Station Id as string */
  stationId: string;

  /** The description */
  description?: string;

  /** The description label color */
  descriptionLabelColor?: string;

  /** Channel Id as string */
  channelId: string;

  /** Boolean is default channel */
  isDefaultChannel: boolean;

  /** Epoch seconds start time */
  displayStartTimeSecs: number;

  /** Boolean is default channel */
  displayEndTimeSecs: number;

  /** Web Workers */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  workerRpcs: any[];

  /** The signal detections */
  signalDetections: WeavessTypes.PickMarker[] | undefined;

  /** The predicted phases */
  predictedPhases: WeavessTypes.PickMarker[] | undefined;

  /** Should the user be able to modify the predicted phase */
  disablePredictedPhaseModification: boolean;

  /** Should the user be able to modify the predicted phase */
  disableSignalDetectionModification: boolean;

  /** The theoretical phase windows */
  theoreticalPhaseWindows: WeavessTypes.TheoreticalPhaseWindow[] | undefined;

  /** Collection of markers */
  markers?: WeavessTypes.Markers;

  /** The selections */
  selections: WeavessTypes.Selections;

  /** (optional) callback events */
  events?: WeavessTypes.ChannelContentEvents;

  /** Configuration for weavess */
  initialConfiguration: WeavessTypes.Configuration;

  /** React elements to render in the content-renderer-content area */
  contentRenderers: React.ReactElement[];

  // Functions

  converters: PositionConverters;

  /** Ref to the html canvas element */
  canvasRef(): HTMLCanvasElement | null;

  /**
   * Call back for toast, which is the notification pop up
   *
   * @param message message to be displayed as a string
   * @param intent (optional) NONE, PRIMARY, WARNING, DANGER ex NONE = "none"
   * @param icon (optional) the way it looks
   * @param timeout (optional) time before message disappears
   */
  toast(message: string, intent?: Intent, icon?: IconName, timeout?: number): void;

  /**
   * Sets the Y axis bounds
   *
   * @param min minimum bound as a number
   * @param max Maximum bound as a number
   */
  setYAxisBounds(min: number, max: number);

  /**
   * @returns current view range as [0,1]
   */
  getViewRange(): [number, number];

  /** Issues a rerender of the graphics */
  renderWaveforms(): void;

  /**
   * get the currently displayed viewTimeInterval
   * (the startTime and endTime of the currently displayed view of the waveforms)
   */
  getCurrentViewRangeInSeconds(): WeavessTypes.TimeRange;

  /**
   * Mouse move event
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   */
  onMouseMove(e: React.MouseEvent<HTMLDivElement>): void;

  /**
   * Mouse down event
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   */
  onMouseDown(e: React.MouseEvent<HTMLDivElement>): void;

  /**
   * onMouseUp event handler
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   */
  onMouseUp(e: React.MouseEvent<HTMLDivElement>): void;

  /**
   * onContextMenu event handler
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   */
  onContextMenu(e: React.MouseEvent<HTMLDivElement>): void;

  /**
   * onKeyDown event handler
   *
   * @param e mouse event as React.MouseEvent<HTMLDivElement>
   * @param timeForMouseXPosition the time for the current mouse position X
   */
  onKeyDown(e: React.KeyboardEvent<HTMLDivElement>): void;

  /**
   * (optional) Updates the measure window
   *
   * @param stationId station id as a string
   * @param channel channel config as a Channel
   * @param startTimeSecs epoch start time secs
   * @param endTimeSecs epoch end time secs
   * @param isDefaultChannel boolean
   * @param removeSelection removed measure window selection div
   */
  updateMeasureWindow?(
    stationId: string,
    channel: WeavessTypes.Channel,
    startTimeSecs: number,
    endTimeSecs: number,
    isDefaultChannel: boolean,
    removeSelection: () => void
  ): void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ContentRendererState {}
