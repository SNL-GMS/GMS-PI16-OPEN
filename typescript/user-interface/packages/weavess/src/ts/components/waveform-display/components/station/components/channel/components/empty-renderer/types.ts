export interface EmptyRendererProps {
  /** Epoch seconds start time */
  displayStartTimeSecs: number;

  /** Boolean is default channel */
  displayEndTimeSecs: number;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyRendererState {}
