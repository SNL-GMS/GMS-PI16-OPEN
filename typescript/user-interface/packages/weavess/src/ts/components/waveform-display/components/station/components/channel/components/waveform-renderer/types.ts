import { WeavessTypes } from '@gms/weavess-core';

export interface WaveformRendererProps {
  /** Epoch seconds start time */
  displayStartTimeSecs: number;

  /** Boolean is default channel */
  displayEndTimeSecs: number;

  /** Id of channel segment */
  channelSegmentId: string;

  /** Collection of channel segments */
  channelSegments: Map<string, WeavessTypes.ChannelSegment>;

  /** the min boundary (x value) in gl units */
  glMin: number;

  /** the max boundary (x value) in gl units */
  glMax: number;

  /** Collection of masks */
  masks?: WeavessTypes.Mask[];

  /** Web Workers */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  workerRpcs: any[];

  /** default min/max range scale for the y-axis */
  defaultRange?: WeavessTypes.Range;

  /** helps identify the channel associated to this renderer */
  channelName: string;

  // Callbacks
  /**
   * Sets the Y axis bounds
   *
   * @param min minimum bound as a number
   * @param max Maximum bound as a number
   */
  setYAxisBounds(min: number, max: number);

  /** Issues a re render */
  renderWaveforms(): void;

  /**
   * Used to look up the position buffer data (a Float32Array formatted like so: x y x y x y ...).
   * Takes the position buffer's id.
   */
  getPositionBuffer?(id: string, startTime: number, endTime: number): Promise<Float32Array>;

  /**
   * Used to look up the Channel Segment Boundaries for a given channel segment by name
   */
  getBoundaries?(
    channelName: string,
    channelSegment: WeavessTypes.ChannelSegment,
    timeRange?: WeavessTypes.TimeRange
  ): Promise<WeavessTypes.ChannelSegmentBoundaries>;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface WaveformRendererState {}

export interface Float32ArrayData {
  /** Color */
  color?: string;

  /** Display type */
  displayType?: WeavessTypes.DisplayType[];

  /** Point size */
  pointSize?: number;

  /** Waveform data */
  float32Array: Float32Array;
}
