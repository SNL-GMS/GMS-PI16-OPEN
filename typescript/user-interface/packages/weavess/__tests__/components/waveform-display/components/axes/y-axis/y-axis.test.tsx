/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable @typescript-eslint/no-magic-numbers */
import * as Enzyme from 'enzyme';
import * as React from 'react';

import { YAxisProps } from '../../../../../../src/ts/components/waveform-display/components/axes/y-axis/types';
import { YAxis } from '../../../../../../src/ts/components/waveform-display/components/axes/y-axis/y-axis';
import { actAndWaitForComponentToPaint } from '../../../../../test-util/test-util';

const props: YAxisProps = {
  heightInPercentage: 100,
  minAmplitude: 0,
  maxAmplitude: 100,
  yAxisTicks: [0, 2, 4, 6, 8, 10]
};

const wrapper = Enzyme.mount(<YAxis {...props} />);
const instance: YAxis = wrapper.find(YAxis).instance() as YAxis;

describe('Weavess X Axis', () => {
  it('to be defined', () => {
    expect(YAxis).toBeDefined();
  });

  it('shallow renders', () => {
    const shallow = Enzyme.shallow(<YAxis {...props} />);
    expect(shallow).toMatchSnapshot();

    const yAxis = new YAxis(props);
    const spy = jest.spyOn(yAxis, 'display');
    yAxis.display();
    expect(spy).toBeCalledTimes(1);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('componentDidUpdate', () => {
    const spy = jest.spyOn(instance, 'componentDidUpdate');

    instance.componentDidUpdate();
    expect(spy).toBeCalledTimes(1);
  });

  it('componentDidCatch', () => {
    const spy = jest.spyOn(instance, 'componentDidCatch');
    instance.componentDidCatch(new Error('error'), { componentStack: undefined });
    expect(spy).toBeCalled();
  });

  it('try to display a negative', () => {
    const mean = Enzyme.mount(
      <YAxis {...props} yAxisTicks={undefined} minAmplitude={0} maxAmplitude={0.05} />
    );
    expect(mean).toMatchSnapshot();
  });

  it('updates when display is called', async () => {
    await actAndWaitForComponentToPaint(wrapper, () => {
      instance.display();
    });
    expect(wrapper).toMatchSnapshot();
  });
});
