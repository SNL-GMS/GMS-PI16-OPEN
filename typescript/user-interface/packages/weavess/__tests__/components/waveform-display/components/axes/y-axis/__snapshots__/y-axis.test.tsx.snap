// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weavess X Axis renders 1`] = `
<YAxis
  heightInPercentage={100}
  maxAmplitude={100}
  minAmplitude={0}
  yAxisTicks={
    Array [
      0,
      2,
      4,
      6,
      8,
      10,
    ]
  }
>
  <div
    className="y-axis"
    style={
      Object {
        "height": "calc(100% - 4px)",
        "margin": "2px 0 2px 0",
      }
    }
  />
</YAxis>
`;

exports[`Weavess X Axis shallow renders 1`] = `
<div
  className="y-axis"
  style={
    Object {
      "height": "calc(100% - 4px)",
      "margin": "2px 0 2px 0",
    }
  }
/>
`;

exports[`Weavess X Axis try to display a negative 1`] = `
<YAxis
  heightInPercentage={100}
  maxAmplitude={0.05}
  minAmplitude={0}
>
  <div
    className="y-axis"
    style={
      Object {
        "height": "calc(100% - 4px)",
        "margin": "2px 0 2px 0",
      }
    }
  />
</YAxis>
`;

exports[`Weavess X Axis updates when display is called 1`] = `
<YAxis
  heightInPercentage={100}
  maxAmplitude={100}
  minAmplitude={0}
  yAxisTicks={
    Array [
      0,
      2,
      4,
      6,
      8,
      10,
    ]
  }
>
  <div
    className="y-axis"
    style={
      Object {
        "height": "calc(100% - 4px)",
        "margin": "2px 0 2px 0",
      }
    }
  />
</YAxis>
`;
